server/cmd/dcrdex/config.go

```diff
@@ -0,0 +1,547 @@
+	}
+	if numNets > 1 {
+		err := fmt.Errorf("both testnet and simnet flags specified")
+		fmt.Fprintln(os.Stderr, err)
+		return loadConfigError(err)
+	}
+
+	// Create the app data directory if it doesn't already exist.
+	err = os.MkdirAll(cfg.AppDataDir, 0700)
+	if err != nil {
+		// Show a nicer error message if it's because a symlink is linked to a
+		// directory that does not exist (probably because it's not mounted).
+		if e, ok := err.(*os.PathError); ok && os.IsExist(err) {
+			if link, lerr := os.Readlink(e.Path); lerr == nil {
+				str := "is symlink %s -> %s mounted?"
+				err = fmt.Errorf(str, e.Path, link)
+			}
+		}
+
+		err := fmt.Errorf("failed to create home directory: %v", err)
+		fmt.Fprintln(os.Stderr, err)
+		return nil, err
+	}
+
+	// If a non-default appdata folder is specified, it may be necessary to
+	// adjust other paths. If these other paths are their defaults, they should
+	// be modifed to look under the non-default appdata directory. If they are
+	// not their defaults, the user-specified values should be used.
+	if defaultAppDataDir != cfg.AppDataDir {
+		if defaultDataDir == cfg.DataDir {
+			cfg.DataDir = filepath.Join(cfg.AppDataDir, defaultDataDirname)
+		}
+		if defaultLogDir == cfg.LogDir {
+			cfg.LogDir = filepath.Join(cfg.AppDataDir, defaultLogDirname)
+		}
+	}
+
+	// Warn about missing config file after the final command line parse
+	// succeeds.  This prevents the warning on help messages and invalid
+	// options.
+	if configFileError != nil {
+		fmt.Printf("%v\n", configFileError)
+		return loadConfigError(configFileError)
+	}
+
+	// Append the network type to the data directory so it is "namespaced" per
+	// network.  In addition to the block database, there are other pieces of
+	// data that are saved to disk such as address manager state. All data is
+	// specific to a network, so namespacing the data directory means each
+	// individual piece of serialized data does not have to worry about changing
+	// names per network and such.
+	//
+	// Make list of old versions of testnet directories here since the network
+	// specific DataDir will be used after this.
+	cfg.DataDir = cleanAndExpandPath(cfg.DataDir)
+	cfg.DataDir = filepath.Join(cfg.DataDir, network.String())
+	// Create the data folder if it does not exist.
+	err = os.MkdirAll(cfg.DataDir, 0700)
+	if err != nil {
+		return nil, err
+	}
+
+	logRotator = nil
+	// Append the network type to the log directory so it is "namespaced"
+	// per network in the same fashion as the data directory.
+	cfg.LogDir = cleanAndExpandPath(cfg.LogDir)
+	cfg.LogDir = filepath.Join(cfg.LogDir, network.String())
+
+	// Ensure that the RPC key and certificate are absolute paths.
+	certDir, certFile := filepath.Split(cfg.RPCCert)
+	if !filepath.IsAbs(certDir) {
+		cfg.RPCCert = filepath.Join(cfg.AppDataDir, certFile)
+	}
+	keyDir, keyFile := filepath.Split(cfg.RPCKey)
+	if !filepath.IsAbs(keyDir) {
+		cfg.RPCKey = filepath.Join(cfg.AppDataDir, keyFile)
```

Comment:
Need to do the same with `DEXPrivKeyPath`. Right now its looking in current working directory.